src/main/diagnostics/diagnostics.ts

```diff
@@ -1,68 +1,121 @@
-import DIAGNOSTICS_TABLE from '@/assets/diagnostics.yaml';
+export enum DiagnosticsCategory {
+  Utilities = 'Utilities',
+  Networking = 'Networking',
+}
+
+/**
+ * DiagnosticsChecker describes an implementation of a single diagnostics check.
+ */
+export interface DiagnosticsChecker {
+  /** Unique identifier for this check. */
+  id: string;
+  documentation: string,
+  description: string,
+  category: DiagnosticsCategory,
+  /**
+   * Preform the check.  Returns true if the check passed (i.e. things are
+   * working correctly and does not need to be fixed).
+   */
+  check(): Promise<boolean>;
+}
 
 type DiagnosticsFix = {
-  fixes: { description: string }
+  description: string
 };
 
-export type DiagnosticsCheck = {
+export type DiagnosticsResult = {
   id: string,
   documentation: string,
   description: string,
-  category: string,
+  category: DiagnosticsCategory,
+  passed: boolean,
   mute: boolean,
   fixes: DiagnosticsFix[],
 };
 
-type DiagnosticsCategories = {
-  title: string,
-  checks: Array<DiagnosticsCheck>,
-};
-
-type DiagnosticsType = {
-  last_update: string
-  categories: Array<DiagnosticsCategories>,
+export type DiagnosticsResultGroup = {
+  last_update: string,
+  checks: Array<DiagnosticsResult>,
 };
 
 export class Diagnostics {
```

Comment:
`DiagnosticsResultGroup` is the data structure that gets sent back over HTTP (via JSON serialization). `Diagnostics` is the active component that produces results.

(I'm happy to change the names, because they're obviously unclear.)

Reply:
Changes made in da002f7c1f0fb21191d9e513689e674546cd5074:
- The class is renamed to `DiagnosticsManager`.
- `DiagnosticsResultGroup` renamed to `DiagnosticsResultCollection`.